src/RobotInterface/src/Helper.cpp

```diff
@@ -489,6 +504,21 @@ bool RobotInterface::setPositionReferences(const iDynTree::VectorDynSize& desire
             return false;
         }
         m_controlMode = VOCAB_CM_POSITION;
+
+        std::vector<yarp::dev::InteractionModeEnum>  modes;
+        std::vector<int>  jointsIndex;
+        for (int i=0;i<m_actuatedDOFs;i++) {
+            jointsIndex.at(i)=i;
+            if(m_isJointModeStiffVector.at(i)==true){
```

Comment:
`m_isJointModeStiffVector.at(i)` is already a bool so you can rewrite this line of the code 
` if(m_isJointModeStiffVector.at(i))`